### ISSUE TITLE
CASE WHEN returns errors

### ISSUE THREAD

**Issue by matthias-Q** [DerekStride/tree-sitter-sql#42]
Hi,
I was trying to find the cause of the following example query:
```sql
WITH test_data AS (
SELECT 
    *
FROM
    (VALUES 
        (true, true, false), 
        (false, true, true), 
        (true, true, true)) 
    AS t (a,b,c)
)
SELECT 
    CASE WHEN a THEN 'A' END AS A,
    CASE WHEN b THEN 'B' END AS B,
    CASE WHEN c THEN 'C' END AS C
FROM test_data;
```

This returns errors in `CASE WHEN a THEN 'A' END AS A` after the when. There must be some error in the second choice on the following section ([L914](https://github.com/DerekStride/tree-sitter-sql/blob/ae3384d69ca86c39dfe81248cb6010e3163bdd65/grammar.js#L914))
```javascript
choice(
  // simplified CASE x WHEN
  seq(
    $._expression,
    $.keyword_when,
    $._expression,
    $.keyword_then,
    $._expression,
    repeat(
      seq(
        $.keyword_when,
        $._expression,
        $.keyword_then,
        $._expression,
      )
    ),
  ),
  // standard CASE WHEN x, where x must be a predicate
  seq(
    $.keyword_when,
    $.predicate,
    $.keyword_then,
    $._expression,
    repeat(
      seq(
        $.keyword_when,
        $.predicate,
        $.keyword_then,
        $._expression,
      )
    ),
  ),
),
```
Can you take a look?

In the second choice, we expect a predicate after the `when`. However, it can also be a boolean column. Can we handle this case at all?

**Comment by dmfay** [comment]
We could turn line 934 into this:

          choice(
            $.predicate,
            $.identifier,
          ),

any problem you can think of with that? We obviously don't know whether the column is a boolean but that part isn't really the parser's job anyway.

**Comment by matthias-Q** [comment]
Ah, yeah you are right. I do not used `$.identifier` when I was testing it locally.

**Comment by DerekStride** [comment]
closed by #43
